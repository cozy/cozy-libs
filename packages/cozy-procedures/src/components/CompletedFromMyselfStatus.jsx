import React from 'react'
import PropTypes from 'prop-types'
import { translate, Bold } from 'cozy-ui/transpiled/react/'

export const CompletedFromMyselfStatus = ({ completed, total, t }) => {
  if (completed === 0) {
    return null
  }

  return (
    <div className="u-mt-1-half u-ml-half u-mr-half u-ta-center">
      <Bold tag="span" className="u-weirdGreen">
        {t('personalDataForm.completedStatus.main', {
          smart_count: completed,
          total
        })}
      </Bold>{' '}
      {t('personalDataForm.completedStatus.rest', { smart_count: completed })}
    </div>
  )
}

CompletedFromMyselfStatus.propTypes = {
  completed: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
  t: PropTypes.func.isRequired
}

export default translate()(CompletedFromMyselfStatus)
