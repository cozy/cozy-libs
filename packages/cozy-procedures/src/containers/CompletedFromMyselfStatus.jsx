import { connect } from 'react-redux'

import context from '../redux/context'
import CompletedFromMyselfStatus from '../components/CompletedFromMyselfStatus'
import {
  getCompletedFromMyself,
  getTotalFields
} from '../redux/personalDataSlice'

export const mapStateToProps = state => ({
  completed: getCompletedFromMyself(state),
  total: getTotalFields(state)
})

export default connect(
  mapStateToProps,
  null,
  null,
  { context }
)(CompletedFromMyselfStatus)
