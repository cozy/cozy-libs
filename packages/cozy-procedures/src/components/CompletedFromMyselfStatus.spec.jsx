import React from 'react'

import { CompletedFromMyselfStatus } from './CompletedFromMyselfStatus'

const fakeT = (msgId, args) => {
  return `${msgId} ${JSON.stringify(args)}`
}

describe('CompletedFromMyselfStatus component', () => {
  it('should display nothing if no data is filled', () => {
    const component = shallow(
      <CompletedFromMyselfStatus completed={0} total={10} t={fakeT} />
    ).getElement()
    expect(component).toBeNull()
  })

  it('should display a status if there are some completed data', () => {
    const component = mount(
      <CompletedFromMyselfStatus completed={5} total={10} t={fakeT} />
    )
    expect(component).toMatchSnapshot()
  })
})
