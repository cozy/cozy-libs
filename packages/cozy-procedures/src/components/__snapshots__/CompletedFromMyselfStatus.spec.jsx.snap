// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedFromMyselfStatus component should display a status if there are some completed data 1`] = `
<CompletedFromMyselfStatus
  completed={5}
  t={[Function]}
  total={10}
>
  <div>
    <span
      className="u-weirdGreen"
    >
      personalDataForm.completedStatus.main {"smart_count":5,"total":10}
    </span>
     
    personalDataForm.completedStatus.rest {"smart_count":5}
  </div>
</CompletedFromMyselfStatus>
`;
